### PR TITLE
added feature: search user using name

### PR DESCRIPTION
**Pull request by StressedByAThousandCodes**

### CODE REVIEW

**Review by carl-howell — APPROVED** [review]

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -27,12 +27,20 @@ export class UserController {
...
+    return this.userService.search(body.name);
+  }
+
   @UseGuards(AuthGuard('jwt'))
   @Get(':id')
   findOneUser(@Param('id') id: number) {
     return this.userService.findUser(id);
   }
 
+
```
Comment (carlofurio-swr) [inline src/user/user.controller.ts]: remove these extra spaces.

File: src/user/user.service.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { DatabaseService } from 'libs/database/database.service';
 import { UpdateUserDto, UserDto, UserPayload } from 'libs/model/user/user.dto';
```
Comment (carlofurio-swr) [inline src/user/user.service.ts]: remove these extra spaces

**Review by carlofurio-swr — CHANGES_REQUESTED** [review]

File: src/user/user.controller.ts [hunk 3]
```diff
@@ -27,12 +27,20 @@ export class UserController {
     return this.userService.findAll();
   }
 
+  @UseGuards(AuthGuard('jwt'))
+  @Get('search')
+  findUserByName(@Body() body: any) {
```
Comment (mika-ella) [inline src/user/user.controller.ts]: create type for body

Comment (mika-ella) [reply src/user/user.controller.ts]: can this be a params? not a body

Comment (StressedByAThousandCodes) [reply src/user/user.controller.ts]: there is an existing same function that has a params
@UseGuards(AuthGuard('jwt'))
  @Get(':id')
  findOneUser(@Param('id') id: number) {
    return this.userService.findUser(id);
  }

**Review by mika-ella — APPROVED** [review]

**Pull request merged**